nghttp/idf_component.yml

```diff
@@ -0,0 +1,5 @@
+version: "1.41.0"
+description: "nghttp2 - HTTP/2 C Library"
+url: https://github.com/espressif/idf-extra-components/tree/master/nghttp
+dependencies:
+  idf: ">=5.0"
```

Comment:
Is this 'just in case' requirement, or we really don't support it with IDF versions 4.x ?

Reply:
We will be splitting `nghttp` from ESP-IDF into 2 component, `http_parser` and `nghttp` (vanilla version). This will be part of ESP-IDF v5.0 and hence have kept this requirement.

Reply:
One minor complication I haven't thought of before, how do we support projects which should be buildable both with 4.x and 5.x, and which depend on this component?

If we add idf_component.yml to support building with IDF 5.0, and use it to install nghttp, then there will be an issue when building the same project with 4.4 — installation will fail.

@kumekay do IDF versions requirements participate in component version resolution?

I.e. can we:
- upload nghttp v0.0.0, which would be a no-op component (no sources, no headers) declared compatible with IDF 4.x
- upload nghttp v1.41.0, which would be the real nghttp component compatible with IDF 5.x
- depending on the user's IDF version, install one or the other.

Reply:
> One minor complication I haven't thought of before, how do we support projects which should be buildable both with 4.x and 5.x, and which depend on this component?

One thing we could do is to backport the change where we split out `http_parser` component to v4.4/v4.3. Then the current PR can have "idf: >=4.3", and the application should build against 4.x and 5.x.

Reply:
Sounds good. We can go ahead with v0.0.0 trick then. Please help me with the upload after this PR gets merged. @kumekay

Reply:
@mahavirj  I tried this approach and found an issue with the component manager that blocks the 0.0.0 trick right now. We will work on the fix

Reply:
At the same time, I don't mind merging this as it is, with support for IDF 5.0+ only. 

> how do we support projects which should be buildable both with 4.x and 5.x,

@igrr  Which projects have this problem?

Reply:
This was a hypothetical issue — sh2lib wasn't use much (it seems) so I don't know if there will be a real project which has this issue. However we should find a general solution for this issue and apply it for all the components we move out in v5.0. Any customer project which should be build-able with both 4.4 and 5.0 might be affected, and currently there are no clear instructions about making the projects compatible with both versions.

Reply:
In this case solution proposed in https://github.com/espressif/idf-extra-components/pull/4#discussion_r778121315 will work the best, however it's not implemented yet on the component manager side

Reply:
Alternatively, if we consider option from https://github.com/espressif/idf-extra-components/pull/4#discussion_r777405139 then that can also work. I have tried building application against v4.4 using same approach, it works. We can consider either of the approaches if requirement comes up.